web/impact/impact/v1/views/mentor_program_office_hour_list_view.py

```diff
@@ -0,0 +1,60 @@
+# MIT License
+# Copyright (c) 2019 MassChallenge, Inc.
+from django.db.models import Value as V
+from django.db.models.functions import Concat
+
+from impact.v1.views.base_list_view import BaseListView
+from impact.v1.helpers import (
+    MentorProgramOfficeHourHelper,
+)
+
+ID_FIELDS = ['mentor_id', 'finalist_id']
+NAME_FIELDS = ['mentor_name', 'finalist_name']
+
+
+class MentorProgramOfficeHourListView(BaseListView):
+    view_name = "office_hour"
+    helper_class = MentorProgramOfficeHourHelper
+
+    def filter(self, qs):
+        if not self.request.query_params.keys():
+            return qs
+
+        if self._has_participant_filter(NAME_FIELDS):
+            return self._filter_by_participant_name(qs)
+
+        if self._has_participant_filter(ID_FIELDS):
+            param_items = self.request.query_params.dict().items()
+            return self._filter_by_participant_id(qs, param_items)
+
+    def _filter_by_participant_name(self, qs):
+        params = self.request.query_params
+        mentor_name = params.get('mentor_name', None)
+        finalist_name = params.get('finalist_name', None)
+
+        if mentor_name:
+            return self._generate_name_query(qs, 'mentor', mentor_name)
+        if finalist_name:
+            return self._generate_name_query(qs, 'finalist', finalist_name)
+        return qs.none()
+
+    def _generate_name_query(self, qs, user, name_value):
```

Comment:
`_filter_by_name` wouldn't work here. There is additional complexity because we are filtering by `full name` which does not exist as a field of the related model where it is to retrieved hence the concatenation of the fields while performing the query